### DISCUSSION TITLE
[v3] legacy_model_binding with some model create by withDefault has error

### DISCUSSION THREAD

**Discussion by fripig** [livewire/livewire#5979] [1. Help] [answered]
I trying upgrade to beta.4 now, but have some question. 

```php
    public function newsData()
    {
        return $this->hasOne(NewsMainData::class, 'news_id', 'news_id')->withDefault([
            'cover_description' => '',
            'headline' => '',
            'show_menu' => false,
        ]);
    }
```
The model still not in db , EloquentModelSynth@loadModel will return null;

then get error like  this

`
TypeError: Livewire\Features\SupportLegacyModels\EloquentModelSynth::setDataOnModel(): Argument #1 ($model) must be of type Illuminate\Database\Eloquent\Model, null given
`

So I add this in EloquentModelSynth@loadModel , this error is gone

```php
if(empty($model)){
                $model = new $class();
 }
```

I still working with other test so can't sure it is ok.

**Comment by joshhanley** [comment, marked as answer]
@fripig thanks for reporting! Any chance you could submit a failing test PR so we can investigate? Thanks!

**Reply by fripig** [reply]
https://github.com/livewire/livewire/pull/6141 done

**Comment by Piket564** [comment]
Ok, I discovered that by changing the way the model requests the query, the error of an empty or missing model in the database disappears.

Like this:

```php

protected function loadModel($meta): ?Model
    {
        $class = $meta['class'];

        // If no alias found, this returns `null`
        $aliasClass = Relation::getMorphedModel($class);

        if (!is_null($aliasClass)) {
            $class = $aliasClass;
        }

        if (isset($meta['key'])) {
            $model = new $class;

            if (isset($meta['connection'])) {
                $model->setConnection($meta['connection']);
            }

            $query = $model->newQueryForRestoration($meta['key']);

            if (isset($meta['relations'])) {
                $query->with($meta['relations']);
            }

            if ($query->first() != null)
                $model = $query->first();
        } else {
            $model = new $class();
        }


        return $model;
    }

```

I was trying to resolve the error:

`Livewire\Features\SupportLegacyModels\EloquentModelSynth::setDataOnModel(): Argument #1 ($model) must be of type Illuminate\Database\Eloquent\Model, null given, called in /var/www/manutenzioni/vendor/livewire/livewire/src/Features/SupportLegacyModels/EloquentModelSynth.php on line 80
`

**Reply by nangirril** [reply]
Thank you! This little change fixed my issue with same error.

**Comment by vasile-rus** [comment]
Hi guys,

now we are migrating our project from V2 -> V3 and we are encountering this issue, We are using V3.5.

Do you know if was fixed or if there is a way to resolve it so we don't get this nasty error?

I saw that was a PR https://github.com/livewire/livewire/pull/8047 that was closed. The solution is fixing the issue but from what I read is not the best way to handle that.

Best regards

**Reply by fripig** [reply]
My own solution is to place that field into a separate variable and convert it directly into an array for transmission. 
This is the minimal modification solution in my situation.
